conans/client/toolchain/cmake/generic.py

```diff
@@ -123,55 +90,28 @@ class CMakeGenericToolchain(CMakeToolchainBase):
         set(CMAKE_CXX_STANDARD {{ cppstd }})
         set(CMAKE_CXX_EXTENSIONS {{ cppstd_extensions }})
         {%- endif %}
+        """)
 
-        # Install prefix
-        {% if install_prefix -%}
-        set(CMAKE_INSTALL_PREFIX {{install_prefix}} CACHE STRING "" FORCE)
-        {%- endif %}
+    _toolchain_tpl = textwrap.dedent("""
+        {% extends 'base_toolchain' %}
 
-        # Variables
-        {% for it, value in variables.items() -%}
-        set({{ it }} "{{ value }}")
-        {%- endfor %}
-        # Variables  per configuration
-        {% for it, values in variables_config.items() -%}
-            {%- set genexpr = namespace(str='') %}
-            {%- for conf, value in values -%}
-                {%- set genexpr.str = genexpr.str +
-                                      '$<IF:$<CONFIG:' + conf + '>,"' + value|string + '",' %}
-                {%- if loop.last %}{% set genexpr.str = genexpr.str + '""' -%}{%- endif -%}
-            {%- endfor -%}
-            {% for i in range(values|count) %}{%- set genexpr.str = genexpr.str + '>' %}
-            {%- endfor -%}
-        set({{ it }} {{ genexpr.str }})
-        {%- endfor %}
-
-        # Preprocessor definitions
-        {% for it, value in preprocessor_definitions.items() -%}
-        # add_compile_definitions only works in cmake >= 3.12
-        add_definitions(-D{{ it }}="{{ value }}")
-        {%- endfor %}
-        # Preprocessor definitions per configuration
-        {% for it, values in preprocessor_definitions_config.items() -%}
-            {%- set genexpr = namespace(str='') %}
-            {%- for conf, value in values -%}
-                {%- set genexpr.str = genexpr.str +
-                                      '$<IF:$<CONFIG:' + conf + '>,"' + value|string + '",' %}
-                {%- if loop.last %}{% set genexpr.str = genexpr.str + '""' -%}{%- endif -%}
-            {%- endfor -%}
-            {% for i in range(values|count) %}{%- set genexpr.str = genexpr.str + '>' %}
-            {%- endfor -%}
-        add_definitions(-D{{ it }}={{ genexpr.str }})
-        {%- endfor %}
+        {% block before_try_compile %}
+        {% include 'before_try_compile' %}
+        {% endblock %}
 
+        {% block main %}
+        {% include 'main' %}
+        {% endblock %}
 
+        {% block footer %}
```

Comment:
Maybe same with the name? Could this section be something like ``cmake_flags_init`` or something similar?